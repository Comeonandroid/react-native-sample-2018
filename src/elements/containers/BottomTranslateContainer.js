import React from 'react'
import styled from 'styled-components'
import { ifProp } from 'styled-tools'
import PropTypes from 'prop-types'

import { View } from '../index'

const BottomTranslateContainer = ({ children, active, ...rest }) => (
  <Element {...rest} active={active}>
    {children}
  </Element>
)

BottomTranslateContainer.propTypes = {
  children: PropTypes.node,
  active: PropTypes.bool
}

BottomTranslateContainer.propTypes = {
  active: false
}

const Element = styled(View) `
  position: absolute;
  bottom: 0;
  width: 100%;
  transform: translateY(${ifProp('active', '0px', '300px')});
`

export default BottomTranslateContainer