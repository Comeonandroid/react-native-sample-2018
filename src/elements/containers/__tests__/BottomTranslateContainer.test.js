import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import BottomTranslateContainer from '../BottomTranslateContainer'

describe('<BottomTranslateContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <BottomTranslateContainer /> })
  itBehavesLike('aComponentWithChildren', { subject: <BottomTranslateContainer /> })

  it('active: flase. Should be hidden at the bottom (translateY > height)', () => {
    const tree = renderer.create(<BottomTranslateContainer active={false} />).toJSON()
    expect(tree.props.style[1][0].transform[0].translateY).toEqual(300)
  })

  it('active: true. Should be hidden at the bottom (translateY = 0)', () => {
    const tree = renderer.create(<BottomTranslateContainer active />).toJSON()
    expect(tree.props.style[1][0].transform[0].translateY).toEqual(0)
  })
})
