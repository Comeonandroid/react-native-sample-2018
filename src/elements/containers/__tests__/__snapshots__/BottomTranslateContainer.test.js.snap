// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomTranslateContainer /> dark: renders correctly 1`] = `
<View
  active={undefined}
  style={
    Array [
      Object {},
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 300,
            },
          ],
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<BottomTranslateContainer /> light: renders correctly 1`] = `
<View
  active={undefined}
  style={
    Array [
      Object {},
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 300,
            },
          ],
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<BottomTranslateContainer /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<BottomTranslateContainer /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
